### PR TITLE
use LF as EOL on windows, adds windows based github actions

### PR DESCRIPTION
**Pull request by clxmstaab**
.. to make the testsuite pass by making line endings consistent across OS

closes https://github.com/phpro/soap-client/issues/381

### CODE REVIEW

**Comment by clxmstaab** [comment]
any ideas on how to handle the remaining test-failures?

image

**Comment by veewee** [comment]
That model uses the `DIRECTORY_SEPARATOR` const:
https://github.com/phpro/soap-client/blob/7e26cd3b6e43542c824fdd4e232fd3d0575c1038/src/Phpro/SoapClient/CodeGenerator/Model/Type.php#L102

We could also use that in the tests as input and expected output.

**Comment by clxmstaab** [comment]
> That model uses the `DIRECTORY_SEPARATOR` const:

ohh.. just using `/` makes it work easily.

remaining problems seems to be some file-stream-resource errors, which are triggered by amphp from the test-runner:

image

**Comment by clxmstaab** [comment]
> Can you make it upgrade grumphp-shim to v1.3.3?

https://github.com/phpro/soap-client/pull/384

**Comment by clxmstaab** [comment]
> Looks like `intl` is missing from the extensions in github actions.

opened a upstream issue, as I think symfony should declare the dependency in composer.json:
https://github.com/symfony/symfony/issues/42089

fixed the github action workflow in the meantime

**Comment by clxmstaab** [comment]
it seems after the grump-php update, the github windows ci is running endless

**Comment by veewee** [comment]
That's strange... Is this something you can reproduce on your local windows machine? I don't own one :)

**Comment by clxmstaab** [comment]
> Is this something you can reproduce on your local windows machine?

```
mstaab@NB-COMPLEX-15 MINGW64 /c/dvl/GitHub/soap-client (win-eol)
$ php vendor/bin/grumphp run --no-interaction
GrumPHP is sniffing your code!

Running tasks with priority 0!
==============================

Running task 1/5: phpcs... ✔
Running task 2/5: phpspec... ✔
Running task 3/5: phpunit... ✔
Running task 4/5: phplint... ✔
Running task 5/5: phpstan...
             ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
           ▄▄▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▌
         ▄▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▌
        ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▌
        ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▌
  ▄▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▌
 ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▌
 ▐█▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▌
   ▀█▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▌
     ▀▀▓▓▓▓▓▓▓▓▓▓▓▓█▀▀▀▀▀▀▀▀▀▀▀▀▀▀████████████▄
      ▄████████▀▀▀▀▀                 ██████████
     ███████▀                         ██████▀
      ▐████      ██▌          ██       ████▌
        ▐█▌                            ███
         █▌           ▄▄ ▄▄           ▐███
        ███       ▄▄▄▄▄▄▄▄▄▄▄▄       ▐███
         ██▄ ▐███████████████████████████
        █▀█████████▌▀▀▀▀▀▀▀▀▀██████████▌▐
          ███████████▄▄▄▄▄▄▄███████████▌
         ▐█████████████████████████████
          █████████████████████████████
           ██ █████████████████████▐██▀
            ▀ ▐███████████████████▌ ▐▀
                ████▀████████▀▐███
                 ▀█▌  ▐█████  ▐█▌
                        ██▀   ▐▀
       _    _ _                         _ _
      / \  | | |   __ _  ___   ___   __| | |
     / _ \ | | |  / _` |/ _ \ / _ \ / _` | |
    / ___ \| | | | (_| | (_) | (_) | (_| |_|
   /_/   \_\_|_|  \__, |\___/ \___/ \__,_(_)
                  |___/

```

works locally for me .. I just re-triggered the CI built.. lets see whether the problem is reproducible

File: .github/workflows/grumphp.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
               with:
                   php-version: ${{ matrix.php-versions }}
                   tools: 'composer:v2'
-                  extensions: pcov, mbstring, posix, soap
+                  extensions: pcov, mbstring, posix, soap, intl
```
Comment (clxmstaab) [inline .github/workflows/grumphp.yml:22]: hmm .. is the posix extension essential?

image

Comment (veewee) [reply .github/workflows/grumphp.yml:22]: No dont think so. Maybe we don't even need pcov ATM, since we are not doing any coverage reporting

**Comment by clxmstaab** [comment]
> It should have been fixed in amphp and compiled into grumphp-shim.
> Can you make it upgrade grumphp-shim to v1.3.3?

maybe you could update grumpphp with the latest dependency updates and create a release?
maybe there is some bugfix release in the depenencies fixing PHP+windows issues?

**Comment by veewee** [comment]
That would rather be a long shot. Maybe we can add a GitHub action in this pr that runs the tools manually, just to see if one of those tools hangs on windows?

It might be a dependency issue from this package as well.

**Comment by staabm** [comment]
@veewee disabling parallelization fixed the problem. Read to merge

File: .github/workflows/grumphp.yml [hunk 2]
```diff
@@ -4,9 +4,10 @@ on: [push, pull_request]
 jobs:
     run:
         runs-on: ${{ matrix.operating-system }}
+        timeout-minutes: 10
```
Comment (staabm) [inline .github/workflows/grumphp.yml:7]: Without this change jobs take 6h before timeing out

**Comment by veewee** [comment]
Thanks for figuring it all out !
Some things could be finetuned in the future, but it looks ok for now.

**Comment by clxmstaab** [comment]
> Some things could be finetuned in the future, but it looks ok for now.

Agree - its not perfect, but a good point to start iterating on

**Pull request merged**